### ISSUE TITLE
Classes become inconsistent if they're first used through a property.

### ISSUE THREAD

**Issue by TerranceN** [godotjs/GodotJS#13]
Using the main branch, and v8 runtime, if I have a script like:

```typescript
export default class Test extends Node3D {
  _ready(): void {
    console.log("new Vector3() instanceof Vector3", new Vector3() instanceof Vector3);
    console.log("new Vector3().constructor === Vector3", new Vector3().constructor === Vector3);
    console.log("this.position instanceof Vector3", this.position instanceof Vector3);
  }
}
```

When loaded, correctly prints:

```
[JS] new Vector3() instanceof Vector3 true
[JS] new Vector3().constructor === Vector3 true
[JS] this.position instanceof Vector3 true
```

But if I first do anything with `this.position`:

```typescript
export default class Test extends Node3D {
  _ready(): void {
    this.position; // Do nothing with it, just reference it
    console.log("new Vector3() instanceof Vector3", new Vector3() instanceof Vector3);
    console.log("new Vector3().constructor === Vector3", new Vector3().constructor === Vector3);
    console.log("this.position instanceof Vector3", this.position instanceof Vector3);
  }
}
```

I incorrectly get:
```
[JS] new Vector3() instanceof Vector3 false
[JS] new Vector3().constructor === Vector3 false
[JS] this.position instanceof Vector3 false
```

And if I just instantiate a Vector3 before both of these:

```typescript
export default class Test extends Node3D {
  _ready(): void {
    new Vector3(); // Do nothing with it, just instantiate it
    this.position; // Do nothing with it, just reference it
    console.log("new Vector3() instanceof Vector3", new Vector3() instanceof Vector3);
    console.log("new Vector3().constructor === Vector3", new Vector3().constructor === Vector3);
    console.log("this.position instanceof Vector3", this.position instanceof Vector3);
  }
}
```

I once again correctly get:

```
[JS] new Vector3() instanceof Vector3 true
[JS] new Vector3().constructor === Vector3 true
[JS] this.position instanceof Vector3 true
```

My guess from this behaviour:

I wonder if accessing `this.position` first is internally causing the creation of a Vector3 type that's somehow different than the Vector3 that's imported? And maybe that's being cached so that subsequent `new Vector3()` calls return that cached class rather than the `Vector3` from the import? And if we do `new Vector3()` first, then that's what gets saved to the cache (and so `this.position` matches the imported `Vector3`)?

**Comment by ialex32x** [comment]
It may caused by the same problem in #12. The prototype.constructor is unexpectedly changed after `ObjectTemplate.GetFunction`, I'm tracing it but haven't got any meaningful clues.

**Comment by ialex32x** [comment]
This issue should also be resolved after #12 is fixed.
